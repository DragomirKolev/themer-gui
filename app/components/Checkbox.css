.input {
  display: none;
}

.label {
  display: flex;
  align-items: center;
}

.labelText {
  margin-left: .31640625em;
}
