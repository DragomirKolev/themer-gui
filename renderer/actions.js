// Actions

export const COLOR_CHANGE = 'COLOR_CHANGE';
export const FOCUS_MODE_TOGGLE = 'FOCUS_MODE_TOGGLE ';
export const EXPORT_DIALOG_OPEN = 'EXPORT_DIALOG_OPEN';
export const TIPS_DIALOG_OPEN = 'TIPS_DIALOG_OPEN';
export const CLOSE_DIALOGS = 'CLOSE_DIALOGS';
export const SET_EXPORT_OPTION = 'SET_EXPORT_OPTION';
export const EXPORT_PROGRESS_DIALOG_OPEN = 'EXPORT_PROGRESS_DIALOG_OPEN';
export const EXPORT_CANCELLED = 'EXPORT_CANCELLED';
export const EXPORT_PROGRESS = 'EXPORT_PROGRESS';
export const EXPORT_ERROR = 'EXPORT_ERROR';
export const EXPORT_COMPLETE = 'EXPORT_COMPLETE';
export const EXPORT_RESET = 'EXPORT_RESET';
export const PREFILL_DIALOG_OPEN = 'PREFILL_DIALOG_OPEN';
export const PREFILL_COLOR_SET_SELECTION_CHANGE = 'PREFILL_COLOR_SET_SELECTION_CHANGE';
export const PREFILL_WITH_COLOR_SET = 'PREFILL_WITH_COLOR_SET';
export const SAVE_COMPLETE = 'SAVE_COMPLETE';
export const OPEN_COMPLETE = 'OPEN_COMPLETE';
export const RESET_STATE = 'RESET_STATE';
export const UPDATE_READY = 'UPDATE_READY';
export const NEXT_TIP = 'NEXT_TIP';
export const PREVIOUS_TIP = 'PREVIOUS_TIP';

// Action generators

export const colorChange = (colorSetKey, colorKey, value) => ({
  type: COLOR_CHANGE,
  colorSetKey,
  colorKey,
  value,
});

export const focusModeToggle = (isLight) => ({
  type: FOCUS_MODE_TOGGLE,
  isLight,
});

export const exportDialogOpen = () => ({
  type: EXPORT_DIALOG_OPEN,
});

export const tipsDialogOpen = () => ({
  type: TIPS_DIALOG_OPEN,
});

export const closeDialogs = () => ({
  type: CLOSE_DIALOGS,
});

export const setExportOption = (option, value) => ({
  type: SET_EXPORT_OPTION,
  option,
  value,
});

export const exportProgressDialogOpen = () => ({
  type: EXPORT_PROGRESS_DIALOG_OPEN,
});

export const exportCancelled = () => ({
  type: EXPORT_CANCELLED,
});

export const exportProgress = (status) => ({
  type: EXPORT_PROGRESS,
  status,
});

export const exportError = (message) => ({
  type: EXPORT_ERROR,
  message,
});

export const exportComplete = (exportedPath) => ({
  type: EXPORT_COMPLETE,
  exportedPath,
});

export const exportReset = () => ({
  type: EXPORT_RESET,
});

export const prefillDialogOpen = () => ({
  type: PREFILL_DIALOG_OPEN,
});

export const prefillColorSetSelectionChange = (selection) => ({
  type: PREFILL_COLOR_SET_SELECTION_CHANGE,
  selection,
});

export const prefillWithColorSet = (colorSet) => ({
  type: PREFILL_WITH_COLOR_SET,
  colorSet,
});

export const saveComplete = (filePath) => ({
  type: SAVE_COMPLETE,
  filePath,
});

export const openComplete = (fileData) => ({
  type: OPEN_COMPLETE,
  fileData,
});

export const resetState = () => ({
  type: RESET_STATE,
});

export const updateReady = () => ({
  type: UPDATE_READY,
});

export const nextTip = () => ({
  type: NEXT_TIP,
});

export const previousTip = () => ({
  type: PREVIOUS_TIP,
});
