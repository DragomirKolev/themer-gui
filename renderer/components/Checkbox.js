import React from 'react';
import { Check } from './Icons';
import css from './InlineInput.css';

export default ({ value, label, onChange }) => (
  <label className={ css.label }>
    <input
      className={ css.input }
      type="checkbox"
      checked={ value }
      onChange={ evt => onChange(evt.target.checked) }
    />
    <Check
      backgroundColor={ value ? 'currentColor' : 'transparent' }
      outlineColor={ value ? 'transparent' : 'currentColor' }
      checkColor={ value ? 'white' : 'transparent' }
    />
    <span className={ css.labelText }>{ label }</span>
  </label>
);
