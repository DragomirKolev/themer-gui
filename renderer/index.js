import React from 'react';
import ReactDom from 'react-dom';
import { Provider } from 'react-redux';
import application from './reducers/reducers';
import { createStore } from 'redux';
import App from './components/App';
import {
  EXPORT_PROGRESS,
  EXPORT_CANCEL,
  EXPORT_ERROR,
  EXPORT_COMPLETE,
  SAVE_COMPLETE,
  OPEN_COMPLETE,
} from '../common/ipcevents';
import {
  closeDialogs,
  exportCancelled,
  exportProgress,
  exportError,
  exportComplete,
} from './actions';
import { ipcRenderer } from 'electron';
import connectMenu from './menu';
import connectUpdater from './updater';
import './index.css';

const store = createStore(application);

// Main events

ipcRenderer.on(EXPORT_CANCEL, () => {
  store.dispatch(closeDialogs());
  store.dispatch(exportCancelled());
});

ipcRenderer.on(EXPORT_PROGRESS, (event, status) => {
  store.dispatch(exportProgress(status));
});

ipcRenderer.on(EXPORT_ERROR, (event, message) => {
  store.dispatch(exportError(message));
});

ipcRenderer.on(EXPORT_COMPLETE, (event, exportedPath) => {
  store.dispatch(exportComplete(exportedPath));
});

// Menu

connectMenu(store);

// Auto-uodate

connectUpdater(store);

// Bootstrap main application

const root = document.createElement('div');
document.body.appendChild(root);
ReactDom.render(
  <Provider store={ store }><App /></Provider>,
  root,
);
