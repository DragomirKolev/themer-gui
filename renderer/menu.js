import { dialog, ipcRenderer, remote } from 'electron';
import fs from 'fs';
import {
  exportDialogOpen,
  tipsDialogOpen,
  prefillDialogOpen,
  saveComplete,
  openComplete,
  resetState,
  helpDialogOpen,
  distributeShades,
} from './actions';
import { EXPORT_COLORS_REQUEST } from '../common/ipcevents';
import { getOrDefault } from './helpers/color';
import {
  isModified,
  save,
  saveAs,
  open,
  promptForIntentToSave,
  showErrorIfError,
} from './helpers/filesystem';
import overwriteShadesWarning from './helpers/overwriteShadesWarning';
const { app, Menu } = remote;

const areAllParseable = inputtedColors => inputtedColors.every(inputtedColor => !!getOrDefault(inputtedColor));

const getExportThemesLabel = (darkCompleted, lightCompleted) => {
  if (darkCompleted && !lightCompleted) {
    return 'Export Colors && Themes (Dark Only)...';
  }
  else if (lightCompleted && !darkCompleted) {
    return 'Export Colors && Themes (Light Only)...';
  }
  else {
    return 'Export Colors && Themes...';
  }
};

const getExportColorsLabel = (darkCompleted, lightCompleted) => {
  if (darkCompleted && !lightCompleted) {
    return 'Export Colors (Dark Only)...';
  }
  else if (lightCompleted && !darkCompleted) {
    return 'Export Colors (Light Only)...';
  }
  else {
    return 'Export Colors...';
  }
};

const setMenu = store => {

  const state = store.getState();
  const darkCompleted = areAllParseable(Object.values(state.colorSets.dark));
  const lightCompleted = areAllParseable(Object.values(state.colorSets.light));
  const isDialogOpen = Object.values(state.dialogsVisibility).some(v => v);
  const hasFilePath = !!state.filePath;
  const hasColorValues = [
    ...Object.values(state.colorSets.dark),
    ...Object.values(state.colorSets.light),
  ].some(Boolean);
  const areDarkShadesDistributable = areAllParseable([state.colorSets.dark.shade0, state.colorSets.dark.shade7]);
  const shouldDarkShadesDistributeWarn = [
    state.colorSets.dark.shade1,
    state.colorSets.dark.shade2,
    state.colorSets.dark.shade3,
    state.colorSets.dark.shade4,
    state.colorSets.dark.shade5,
    state.colorSets.dark.shade6,
  ].some(Boolean);
  const areLightShadesDistributable = areAllParseable([state.colorSets.light.shade0, state.colorSets.light.shade7]);
  const shouldLightShadesDistributeWarn = [
    state.colorSets.light.shade1,
    state.colorSets.light.shade2,
    state.colorSets.light.shade3,
    state.colorSets.light.shade4,
    state.colorSets.light.shade5,
    state.colorSets.light.shade6,
  ].some(Boolean);

  const template = [
    process.platform === 'darwin' ? {
      label: app.getName(),
      submenu: [
        {role: 'about'},
        {type: 'separator'},
        {role: 'services', submenu: []},
        {type: 'separator'},
        {role: 'hide'},
        {role: 'hideothers'},
        {role: 'unhide'},
        {type: 'separator'},
        {role: 'quit'},
      ],
    } : undefined,
    {
      label: 'File',
      submenu: [
        {
          label: 'New',
          accelerator: 'CmdOrCtrl+N',
          click () {
            const { filePath, ...data } = state;
            if (hasFilePath) {
              isModified(filePath, data)
                .then(modified => {
                  if (modified) {
                    // There was a file path, and were modifications; prompting for save.
                    promptForIntentToSave()
                      .then(wouldLikeToSave => {
                        if (wouldLikeToSave) {
                          return save(filePath, data)
                            .then(pathWritten => store.dispatch(saveComplete(pathWritten)));
                        }
                        else { return Promise.resolve(); }
                      })
                      .then(() => store.dispatch(resetState()))
                      .catch(showErrorIfError);
                  }
                  else {
                    // There was a file path, but no modifications; resetting state.
                    store.dispatch(resetState());
                  }
                });
            }
            else {
              if (hasColorValues) {
                // No file path, but has color values; prompting for save as.
                promptForIntentToSave()
                  .then(wouldLikeToSave => {
                    if (wouldLikeToSave) {
                      return saveAs(data)
                        .then(pathWritten => store.dispatch(saveComplete(pathWritten)));
                    }
                    else { return Promise.resolve(); }
                  })
                  .then(() => store.dispatch(resetState()))
                  .catch(showErrorIfError);
              }
              else {
                // No file path, but no values; resetting state.
                store.dispatch(resetState());
              }
            }
          },
        },
        {
          label: `Save${!hasFilePath ? '...' : ''}`,
          accelerator: 'CmdOrCtrl+S',
          click () {
            const { filePath, ...data } = state;
            if (!!filePath) {
              save(filePath, data).then(pathWritten => store.dispatch(saveComplete(pathWritten)));
            }
            else {
              saveAs(data).then(pathWritten => store.dispatch(saveComplete(pathWritten)));
            }
          },
        },
        {
          label: 'Save As...',
          accelerator: 'CmdOrCtrl+Shift+S',
          click () {
            const { filePath, ...data } = state;
            saveAs(data).then(pathWritten => store.dispatch(saveComplete(pathWritten)));
          },
        },
        {
          label: 'Open...',
          accelerator: 'CmdOrCtrl+O',
          click() {
            const { filePath, ...data } = state;
            if (hasFilePath) {
              isModified(filePath, data)
                .then(modified => {
                  if (modified) {
                    // There was a file path, and were modifications; prompting for save.
                    promptForIntentToSave()
                      .then(wouldLikeToSave => {
                        if (wouldLikeToSave) {
                          return save(filePath, data)
                            .then(pathWritten => store.dispatch(saveComplete(pathWritten)));
                        }
                        else { return Promise.resolve(); }
                      })
                      .then(open)
                      .then(fileData => store.dispatch(openComplete(fileData)))
                      .catch(showErrorIfError);
                  }
                  else {
                    // There was a file path, but no modifications; opening.
                    open()
                      .then(fileData => store.dispatch(openComplete(fileData)))
                      .catch(showErrorIfError);
                  }
                });
            }
            else {
              if (hasColorValues) {
                // No file path, but has color values; prompting for save as.
                promptForIntentToSave()
                  .then(wouldLikeToSave => {
                    if (wouldLikeToSave) {
                      return saveAs(data)
                        .then(pathWritten => store.dispatch(saveComplete(pathWritten)));
                    }
                    else { return Promise.resolve(); }
                  })
                  .then(open)
                  .then(fileData => store.dispatch(openComplete(fileData)))
                  .catch(showErrorIfError);
              }
              else {
                // No file path, but no values; opening.
                open()
                  .then(fileData => store.dispatch(openComplete(fileData)))
                  .catch(showErrorIfError);
              }
            }
          },
        },
        {
          label: 'Prefill With Built-in Color Set...',
          accelerator: 'CmdOrCtrl+Shift+O',
          enabled: !isDialogOpen,
          click () { store.dispatch(prefillDialogOpen()); },
        },
        {type: 'separator'},
        {
          label: getExportThemesLabel(darkCompleted, lightCompleted),
          accelerator: 'CmdOrCtrl+E',
          enabled: darkCompleted || lightCompleted,
          click () { store.dispatch(exportDialogOpen()); },
        },
        {
          label: getExportColorsLabel(darkCompleted, lightCompleted),
          accelerator: 'CmdOrCtrl+Shift+E',
          enabled: darkCompleted || lightCompleted,
          click () { ipcRenderer.send(EXPORT_COLORS_REQUEST, state.colorSets); },
        },
      ],
    },
    {
      label: 'Edit',
      submenu: [
        { role: 'undo' },
        { role: 'redo' },
        { type: 'separator' },
        { role: 'cut' },
        { role: 'copy' },
        { role: 'paste' },
        { role: 'delete' },
        { role: 'selectall' },
        { type: 'separator' },
        {
          label: 'Distribute Shades',
          enabled: areDarkShadesDistributable || areLightShadesDistributable,
          accelerator: 'CmdOrCtrl+Alt+Shift+D',
          click () {
            if (shouldDarkShadesDistributeWarn || shouldLightShadesDistributeWarn) {
              overwriteShadesWarning()
                .then(() => {
                  store.dispatch(distributeShades());
                })
                .catch(() => {});
            }
            else {
              store.dispatch(distributeShades());
            }
          },
        },
      ],
    },
    {
      label: 'View',
      submenu: [
        { role: 'resetzoom' },
        { role: 'zoomin' },
        { role: 'zoomout' },
        { type: 'separator' },
        { role: 'togglefullscreen' },
      ],
    },
    {
      role: 'window',
      submenu: [
        { role: 'minimize' },
        { role: 'close' },
      ],
    },
    {
      role: 'help',
      submenu: [
        {
          label: 'Show Tips...',
          enabled: !isDialogOpen,
          click () { store.dispatch(tipsDialogOpen()); },
        },
        {
          label: 'Show Color Mappings...',
          enabled: !isDialogOpen,
          click() { store.dispatch(helpDialogOpen()); },
        },
      ],
    },
  ].filter(Boolean);
  Menu.setApplicationMenu(Menu.buildFromTemplate(template));
};

export default store => {
  setMenu(store);
  store.subscribe(() => setMenu(store));
};
