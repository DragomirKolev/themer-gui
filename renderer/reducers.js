import { combineReducers } from 'redux';
import {
  COLOR_CHANGE,
  FOCUS_MODE_TOGGLE,
  EXPORT_DIALOG_OPEN,
  HELP_DIALOG_OPEN,
  CLOSE_DIALOGS,
  SET_EXPORT_OPTION,
  EXPORT_PROGRESS_DIALOG_OPEN,
  EXPORT_CANCELLED,
  EXPORT_PROGRESS,
  EXPORT_ERROR,
  EXPORT_COMPLETE,
} from './actions';
import {
  NOT_STARTED,
  IN_PROGRESS,
  COMPLETE_ERROR,
  COMPLETE_SUCCESS,
} from './helpers/exportProgressStates';

const defaultColorSets = {
  dark: {
    shade0: '',
    shade1: '',
    shade2: '',
    shade3: '',
    shade4: '',
    shade5: '',
    shade6: '',
    shade7: '',
    accent0: '',
    accent1: '',
    accent2: '',
    accent3: '',
    accent4: '',
    accent5: '',
    accent6: '',
    accent7: '',
  },
  light: {
    shade0: '',
    shade1: '',
    shade2: '',
    shade3: '',
    shade4: '',
    shade5: '',
    shade6: '',
    shade7: '',
    accent0: '',
    accent1: '',
    accent2: '',
    accent3: '',
    accent4: '',
    accent5: '',
    accent6: '',
    accent7: '',
  },
};

const colorSetsReducer = (state = defaultColorSets, action) => {
  switch (action.type) {
    case COLOR_CHANGE:
      const { colorSetKey, colorKey, value } = action;
      const ret = {
        ...state,
        [colorSetKey]: {
          ...state[colorSetKey],
          [colorKey]: value,
        },
      };
      return ret;
    default:
      return state;
  }
};

const defaultFocusMode = {
  dark: false,
  light: false,
};

const focusModeReducer = (state = defaultFocusMode, action) => {
  switch (action.type) {
    case FOCUS_MODE_TOGGLE:
      const key = action.isLight ? 'light' : 'dark';
      return {
        ...state,
        [key]: !state[key],
      };
    default:
      return state;
  }
};

const defaultDialogsVisibility = {
  export: false,
  help: false,
  exportProgress: false,
};

const dialogsVisibilityReducer = (state = defaultDialogsVisibility, action) => {
  switch (action.type) {
    case EXPORT_DIALOG_OPEN:
      return {
        ...state,
        export: true,
      };
    case HELP_DIALOG_OPEN:
      return {
        ...state,
        help: true,
      };
    case EXPORT_PROGRESS_DIALOG_OPEN:
      return {
        ...state,
        exportProgress: true,
      };
    case CLOSE_DIALOGS:
      return Object.keys(state).reduce((reduced, key) => ({ ...reduced, [key]: false }), {});
    default:
      return state;
  }
};

const defaultExportOptions = {
  hyper: false,
  iterm: false,
  terminal: false,
  atomSyntax: false,
  sublimeText: false,
  vim: false,
  vimLightline: false,
  wallpaperBlockWave: false,
  wallpaperOctagon: false,
  slack: false,
};

const exportOptionsReducer = (state = defaultExportOptions, action) => {
  switch (action.type) {
    case SET_EXPORT_OPTION:
      return {
        ...state,
        [action.option]: action.value,
      };
    default:
      return state;
  }
};

const defaultExportProgress = {
  status: '',
  state: NOT_STARTED,
  exportedPath: null,
};

const exportProgressReducer = (state = defaultExportProgress, action) => {
  switch (action.type) {
    case EXPORT_CANCELLED:
      return defaultExportProgress;
    case EXPORT_PROGRESS:
      return {
        ...defaultExportProgress,
        status: action.status,
        state: IN_PROGRESS,
      };
    case EXPORT_ERROR:
      return {
        ...defaultExportProgress,
        status: `${action.message} ðŸ˜ž`,
        state: COMPLETE_ERROR,
      };
    case EXPORT_COMPLETE:
      return {
        status: `Your themes have been exported to ${action.exportedPath} ðŸŽ‰`,
        state: COMPLETE_SUCCESS,
        exportedPath: action.exportedPath,
      };
    default:
      return state;
  }
};

export default combineReducers({
  colorSets: colorSetsReducer,
  focusMode: focusModeReducer,
  dialogsVisibility: dialogsVisibilityReducer,
  exportOptions: exportOptionsReducer,
  exportProgress: exportProgressReducer,
});
