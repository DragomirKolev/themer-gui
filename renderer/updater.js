import { ipcRenderer } from 'electron';
import {
  CHECK_FOR_UPDATES,
  UPDATE_READY,
} from '../common/ipcevents';
import { updateReady } from './actions';

export default store => {
  ipcRenderer.on(UPDATE_READY, () => {
    store.dispatch(updateReady());
  });
  if (process.env.NODE_ENV !== 'development') {
    ipcRenderer.send(CHECK_FOR_UPDATES);
  }
};
