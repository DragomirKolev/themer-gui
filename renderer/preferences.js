import { defaultPreferences } from './reducers/preferences';

const PREF_KEY = 'preferences';

export const readPreferences = () => {
  try {
    return {
      ...defaultPreferences,
      ...JSON.parse(window.localStorage.getItem(PREF_KEY)),
    };
  }
  catch (e) {
    return defaultPreferences;
  }
};

const writePrefs = prefs => window.localStorage.setItem(PREF_KEY, JSON.stringify(prefs));

export const getPreference = key => readPreferences()[key];

export const connectPreferences = store => {
  store.subscribe(() => {
    writePrefs(store.getState().preferences);
  });
};
