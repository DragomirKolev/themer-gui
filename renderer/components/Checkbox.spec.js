import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
  it('should render properly', () => {
    expect(renderer.create(
      <Checkbox
        value={ false }
        label="off"
        onChange={ () => {} }
      />
    )).toMatchSnapshot();
    expect(renderer.create(
      <Checkbox
        value={ true }
        label="on"
        onChange={ () => {} }
      />
    )).toMatchSnapshot();
  });
});
