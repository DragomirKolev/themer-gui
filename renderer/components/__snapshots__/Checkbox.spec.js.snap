// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render properly 1`] = `
<label
  className="label"
>
  <input
    checked={false}
    className="input"
    disabled={undefined}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    height="16"
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": "2px",
        "boxShadow": "inset 0 0 0 1px currentColor",
      }
    }
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 9l2 2M6 11l6-6"
      fill="none"
      fillRule="evenodd"
      stroke="transparent"
      strokeLinecap="square"
      strokeWidth="2"
    />
  </svg>
  <span
    className="labelText"
  >
    off
  </span>
</label>
`;

exports[`Checkbox should render properly 2`] = `
<label
  className="label"
>
  <input
    checked={true}
    className="input"
    disabled={undefined}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    height="16"
    style={
      Object {
        "backgroundColor": "currentColor",
        "borderRadius": "2px",
        "boxShadow": "inset 0 0 0 1px transparent",
      }
    }
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 9l2 2M6 11l6-6"
      fill="none"
      fillRule="evenodd"
      stroke="white"
      strokeLinecap="square"
      strokeWidth="2"
    />
  </svg>
  <span
    className="labelText"
  >
    on
  </span>
</label>
`;

exports[`Checkbox should render properly 3`] = `
<label
  className="label disabled"
>
  <input
    checked={false}
    className="input"
    disabled={true}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    height="16"
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": "2px",
        "boxShadow": "inset 0 0 0 1px currentColor",
      }
    }
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 9l2 2M6 11l6-6"
      fill="none"
      fillRule="evenodd"
      stroke="transparent"
      strokeLinecap="square"
      strokeWidth="2"
    />
  </svg>
  <span
    className="labelText"
  >
    off
  </span>
</label>
`;

exports[`Checkbox should render properly 4`] = `
<label
  className="label disabled"
>
  <input
    checked={true}
    className="input"
    disabled={true}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    height="16"
    style={
      Object {
        "backgroundColor": "currentColor",
        "borderRadius": "2px",
        "boxShadow": "inset 0 0 0 1px transparent",
      }
    }
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 9l2 2M6 11l6-6"
      fill="none"
      fillRule="evenodd"
      stroke="white"
      strokeLinecap="square"
      strokeWidth="2"
    />
  </svg>
  <span
    className="labelText"
  >
    on
  </span>
</label>
`;
